routeDetails.html

```diff
@@ -0,0 +1,101 @@
+
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+	<style>
+		table {background-color: linen; font-family: "Verdana", Times, serif;}
+		th {background-color: #4CAF50;color: white;}
+		tr:nth-child(even) {background-color: #f2f2f2}
+		p {font-family: "Verdana", Times, serif;}
+	</style>
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+	<title>ArcTouch Code Challenge</title>
+	<script type="text/javascript" src="prototype.js"></script>
+	<script type="text/javascript" src="utils.js"></script>
+	<script type="text/javascript">
+		//<![CDATA[
+
+		var routeId   = getURLParameterByName("routeId");
+		var routeNo   = getURLParameterByName("routeNo");
+		var routeLine = getURLParameterByName("routeLine");
+		var street    = getURLParameterByName("street");
+
+		function getRouteDetails() {
+			var oOptions1 = {
+				method: "POST",
+				postBody: "\n{\n\n\"params\": {\n\n\"routeId\": " + routeId + "\n\n}\n\n}",
+				contentType: "application/json",
+				requestHeaders: [ "X-AppGlu-Environment", "staging", "Authorization", "Basic V0tENE43WU1BMXVpTThWOkR0ZFR0ek1MUWxBMGhrMkMxWWk1cEx5VklsQVE2OA==", "Accept", "application/json"],
+				onSuccess: function (oXHR, oJson) {
+					displayRouteStops(oXHR.responseText);
+				},
+				onFailure: function (oXHR, oJson) {
+					alert("An error occurred requesting route stops: " + oXHR.statusText);
+				}
+			};
+			var oRequest = new Ajax.Request("https://api.appglu.com/v1/queries/findStopsByRouteId/run", oOptions1);
+
+			var oOptions2 = {
+				method: "POST",
+				postBody: "\n{\n\n\"params\": {\n\n\"routeId\": " + routeId + "\n\n}\n\n}",
+				contentType: "application/json",
+				requestHeaders: [ "X-AppGlu-Environment", "staging", "Authorization", "Basic V0tENE43WU1BMXVpTThWOkR0ZFR0ek1MUWxBMGhrMkMxWWk1cEx5VklsQVE2OA==", "Accept", "application/json"],
+				onSuccess: function (oXHR, oJson) {
+					displayRouteDepartures(oXHR.responseText);
+				},
+				onFailure: function (oXHR, oJson) {
+					alert("An error occurred requesting route departures: " + oXHR.statusText);
+				}
+			};
+			var oRequest = new Ajax.Request("https://api.appglu.com/v1/queries/findDeparturesByRouteId/run", oOptions2);

```

Comment:
New function added getRouteAPIResponse(url, type) so now I can call it
twice and avoid the repetition of XMLHttpRequest object. Code changed an
committed. Thanks.

2016-03-02 17:26 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In routeDetails.html
> <https://github.com/otaviofontana/arcTouchOtavioFontana/pull/1#discussion_r54785345>
> :
>
> > +			};
> > +			var oRequest = new Ajax.Request("https://api.appglu.com/v1/queries/findStopsByRouteId/run", oOptions1);
> > +
> > +			var oOptions2 = {
> > +				method: "POST",
> > +				postBody: "\n{\n\n\"params\": {\n\n\"routeId\": " + routeId + "\n\n}\n\n}",
> > +				contentType: "application/json",
> > +				requestHeaders: [ "X-AppGlu-Environment", "staging", "Authorization", "Basic V0tENE43WU1BMXVpTThWOkR0ZFR0ek1MUWxBMGhrMkMxWWk1cEx5VklsQVE2OA==", "Accept", "application/json"],
> > +				onSuccess: function (oXHR, oJson) {
> > +					displayRouteDepartures(oXHR.responseText);
> > +				},
> > +				onFailure: function (oXHR, oJson) {
> > +					alert("An error occurred requesting route departures: " + oXHR.statusText);
> > +				}
> > +			};
> > +			var oRequest = new Ajax.Request("https://api.appglu.com/v1/queries/findDeparturesByRouteId/run", oOptions2);
>
> This code is almost identical to the other network calls. How could you
> avoid this repetition?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/otaviofontana/arcTouchOtavioFontana/pull/1/files#r54785345>
> .
>